packages/feeds-client/src/utils/throttling/throttled-get-batched-own-capabilities.test.ts

```diff
@@ -0,0 +1,68 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import {
+  queueBatchedOwnCapabilities,
+  clearQueuedFeeds,
+} from './throttled-get-batched-own-capabilities';
+
+describe('queueBatchedOwnCapabilities', () => {
+  beforeEach(() => {
+    clearQueuedFeeds();
+  });
+
+  it('calls throttledGetBatchOwnCapabilities with queued feeds and clears them via callback', () => {
+    const throttledGetBatchOwnCapabilities = vi.fn();
+    const client: any = { throttledGetBatchOwnCapabilities };
+
+    // First enqueue two feeds → should call with both
+    queueBatchedOwnCapabilities.call(client, { feeds: ['feed:1', 'feed:2'] });
+    expect(throttledGetBatchOwnCapabilities).toHaveBeenCalledTimes(1);
+
+    const firstArgs = throttledGetBatchOwnCapabilities.mock.calls[0];
+    const firstQueued = firstArgs[0] as string[];
+    const firstCallback = firstArgs[1] as (feedsToClear: string[]) => void;
+    expect(new Set(firstQueued)).toEqual(new Set(['feed:1', 'feed:2']));
+
+    // Simulate downstream completion that clears only `feed:1`
+    firstCallback(['feed:1']);
+
+    // Calling again without adding new feeds should pass only the remaining queued feed
+    queueBatchedOwnCapabilities.call(client, { feeds: [] });
+    expect(throttledGetBatchOwnCapabilities).toHaveBeenCalledTimes(2);
+
+    const secondArgs = throttledGetBatchOwnCapabilities.mock.calls[1];
+    const secondQueued = secondArgs[0] as string[];
+    const secondCallback = secondArgs[1] as (feedsToClear: string[]) => void;
+    expect(secondQueued).toEqual(['feed:2']);
+
+    // Now clear the last remaining feed
+    secondCallback(['feed:2']);
+
+    // With nothing queued anymore, a subsequent call should not trigger a throttled call
+    queueBatchedOwnCapabilities.call(client, { feeds: [] });
+    expect(throttledGetBatchOwnCapabilities).toHaveBeenCalledTimes(2);
+  });
+
+  it('should provide no more feeds than the API limit (100)', () => {
+    const throttledGetBatchOwnCapabilities = vi.fn();
+    const client: any = { throttledGetBatchOwnCapabilities };
+
+    queueBatchedOwnCapabilities.call(client, {
+      feeds: Array.from({ length: 101 }, (_, i) => `feed:${i + 1}`),
+    });
+    expect(throttledGetBatchOwnCapabilities).toHaveBeenCalledTimes(1);
+    const args = throttledGetBatchOwnCapabilities.mock.calls[0];
+
+    expect(args[0]).toHaveLength(100);
```

Comment:
Maybe we should also check which feeds it was initially called with ? As in, their IDs ?

I believe they should be sorted in a `Set()`

Reply:
hm, I thought set is not sorted, that's why I didn't check it here